### ISSUE TITLE
[wishlist] bring support for "stats" atlases awhile back developed by NeuroDebian

### ISSUE THREAD

**Issue by yarikoptic** [pauldmccarthy/fsleyes#28]
Awhile back I have [provided a patch to fslview](https://salsa.debian.org/neurodebian-team/fslview/blob/master/debian/patches/enh_statistics_atlas) which we kept maintaining and delivering patched fslview through NeuroDebian. I shared it on FSL mailing list with

<details>
<summary>The last email on the list in 2015 </summary> 

```shell
Date: Wed, 11 Feb 2015 10:48:39 -0500
From: Yaroslav Halchenko <lists@onerussian.com>
To: FSL@JISCMAIL.AC.UK
Subject: Re: [FSL] fslview -- patch to support "Statistic" atlases

On Tue, 26 Nov 2013, Yaroslav Halchenko wrote:
> >    Agreed. We have Yaroslav's code and are performing some testing. I expect
> >    it to be in the next release of fslview.

> which is coming on .... ? ;)

Since more people start to use neurosynth atlas(es) by now I would like
to bring this thread from the dead.  ATM in NeuroDebian we ship fslview
4.0.1 (from 2013) with those patches on top.  And wouldn't it be
great if they were adopted upstream? ;)

Could someone please update me on the timeline of perspective
fslview releases?

Thank you in advance!
```
</details>

Here you could find tarballs/packages with neurosynth atlas: http://neuro.debian.net/debian/pool/main/n/neurosynth-data/  which are still available from neurodebian.

It would be nice if fsleyes supported such atlases (as of 0.15.2-1~nd+1 I have from NeuroDebian it doesn't yet)

**Comment by pauldmccarthy** [comment]
These do actually seem to load in a more recent version of FSLeyes (I don't manage the neurodebian versions):

![image](https://user-images.githubusercontent.com/236128/62726587-7f07bc80-ba0f-11e9-8ed3-1bfb67155f66.png)

It works as a by-product of the way that the `fsl.data.atlases` module distinguishes between "label" and "probabilistic" atlas types :) 

https://github.com/pauldmccarthy/fslpy/blob/master/fsl/data/atlases.py#L554

I'm happy to add more formal support for `Statistic` atlas types. Could you save me a bit of time, and give me an overview of the differences between `Statistic` and `Probabilistic` atlases?

**Comment by yarikoptic** [comment]
Coolio!  I guess the best description for the differences would be the header in that patch: https://salsa.debian.org/neurodebian-team/fslview/blob/master/debian/patches/enh_statistics_atlas  . I am actually proud of myself how descriptive I was, which should save us both time now.  One immediate difference I see is that since they are not "population probability", there is no `%`.  So it seems primarily to remove assumptions embedded in atlases having solely a population statistic

**Comment by pauldmccarthy** [comment]
@yarikoptic could you have a play with 0.31.0? The only difference to the end user will be the unit reporting, and automatic display range set up

**Comment by yarikoptic** [comment]
Seems to work but displays bunch of -0.00 for anything which shouldn't be displayed I guess:

![screenshot](http://www.onerussian.com/tmp/gkrellShoot_09-10-19_135124.png)

the same if  overlay is in mni.

<details>
<summary>some details on the warnings etc</summary> 

my changes
```shell
$> git diff
diff --git a/fsleyes/gl/__init__.py b/fsleyes/gl/__init__.py
index ac6b8735..6c62b611 100644
--- a/fsleyes/gl/__init__.py
+++ b/fsleyes/gl/__init__.py
@@ -903,7 +903,7 @@ as the meta-class.  This is not necessary under Python2/wxPython.
 
 if fslplatform.wxFlavour == fslplatform.WX_PHOENIX:
 
-    import wx.siplib as sip
+    import sip
     WXGLMetaClass = sip.wrappertype
 
 else:
diff --git a/fsleyes/panel.py b/fsleyes/panel.py
old mode 100644
new mode 100755
index c54d48b7..ce6dfe7f
--- a/fsleyes/panel.py
+++ b/fsleyes/panel.py
@@ -353,7 +353,7 @@ FSLeyesPanelMeta = None
 # wxPython/Phoenix
 if fslplatform.wxFlavour == fslplatform.WX_PHOENIX:
 
-    import wx.siplib as sip
+    import sip
 
     FSLeyesPanelMeta = sip.wrappertype
     FSLeyesPanelBase = wx.Panel
```
and then starting it leads to following
```
$> fsleyes sub-sid000007_ses-actions1_acq-MPRAGE_T1w.nii.gz
  from collections import Mapping
  cursors.MOVE          : Gdk.Cursor.new(Gdk.CursorType.FLEUR),
  "Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
  _dollar_pattern = re.compile("(.*?)\$(\$?[\w\.]+)")
  comment_line_re = re.compile('^\s*\#')
  """Join lines following explicit line continuations (\)"""
  cellmagic_help_re = re.compile('%%\w+\?')
  """, re.VERBOSE)
  pre = re.match('^(\s*)(.*)',line).groups()[0]
  cookie_re = re.compile("coding[:=]\s*([-\w.]+)")
  '''),
  reg = re.compile('^\w+\.\w+$')
  re.match('(pinfo )?(\?*)(.*?)(\??$)',parameter_s).groups()
(fsleyes:13427): Gtk-WARNING **: 13:53:11.819: Negative content width -19 (allocation 1, extents 10x10) while allocating gadget (node button, owner GtkToggleButton)
(fsleyes:13427): Gtk-WARNING **: 13:53:11.819: Negative content height -9 (allocation 1, extents 5x5) while allocating gadget (node button, owner GtkToggleButton)
(fsleyes:13427): Gtk-WARNING **: 13:53:11.819: Negative content width -19 (allocation 1, extents 10x10) while allocating gadget (node button, owner GtkToggleButton)
(fsleyes:13427): Gtk-WARNING **: 13:53:11.819: Negative content height -9 (allocation 1, extents 5x5) while allocating gadget (node button, owner GtkToggleButton)
(fsleyes:13427): Gtk-WARNING **: 13:53:11.844: Negative content width -19 (allocation 1, extents 10x10) while allocating gadget (node button, owner GtkButton)
(fsleyes:13427): Gtk-WARNING **: 13:53:11.844: Negative content height -9 (allocation 1, extents 5x5) while allocating gadget (node button, owner GtkButton)
(fsleyes:13427): Gtk-WARNING **: 13:53:11.844: Negative content width -19 (allocation 1, extents 10x10) while allocating gadget (node button, owner GtkToggleButton)
(fsleyes:13427): Gtk-WARNING **: 13:53:11.844: Negative content height -9 (allocation 1, extents 5x5) while allocating gadget (node button, owner GtkToggleButton)
..... with many more of similar warnings
```

</details>

**Comment by pauldmccarthy** [comment]
Thanks - I'll look at filtering out the `0.00` entries in the next version.

Those warnings look cryptic because I've added [stdout filtering](../blob/master/fsleyes/filtermain.py) to suppress a large number of warnings that come from GTK - what you're seeing are the lines that made it through the filter :)

You could run `fsleyes_unfiltered` to see how much garbage gets printed. Some of them may be deprecation warnings that I have not yet gotten around to fixing, but the majority come from sources which are out of my control.

**Comment by yarikoptic** [comment]
<details>
<summary>FWIW and FTR here is unfiltered output</summary> 

```shell
(git)hopa:~exppsy/fsleyes[yarik]
$> fsleyes_unfiltered 
/usr/lib/python3/dist-packages/jinja2/runtime.py:318: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Mapping
/usr/lib/python3/dist-packages/matplotlib/backends/backend_gtk3.py:28: DeprecationWarning: Gdk.Cursor.new is deprecated
  cursors.MOVE          : Gdk.Cursor.new(Gdk.CursorType.FLEUR),
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1309: DeprecationWarning: NewId() is deprecated
  _NTB_AXISMENU = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1310: DeprecationWarning: NewId() is deprecated
  _NTB_AXISMENU_BUTTON = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1311: DeprecationWarning: NewId() is deprecated
  _NTB_X_PAN_LEFT = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1312: DeprecationWarning: NewId() is deprecated
  _NTB_X_PAN_RIGHT = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1313: DeprecationWarning: NewId() is deprecated
  _NTB_X_ZOOMIN = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1314: DeprecationWarning: NewId() is deprecated
  _NTB_X_ZOOMOUT = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1315: DeprecationWarning: NewId() is deprecated
  _NTB_Y_PAN_UP = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1316: DeprecationWarning: NewId() is deprecated
  _NTB_Y_PAN_DOWN = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1317: DeprecationWarning: NewId() is deprecated
  _NTB_Y_ZOOMIN = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1318: DeprecationWarning: NewId() is deprecated
  _NTB_Y_ZOOMOUT = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1320: DeprecationWarning: NewId() is deprecated
  _NTB_SAVE = wx.NewId()
/usr/lib/python3/dist-packages/matplotlib/backends/backend_wx.py:1321: DeprecationWarning: NewId() is deprecated
  _NTB_CLOSE = wx.NewId()
/usr/lib/python3/dist-packages/IPython/utils/path.py:207: DeprecationWarning: invalid escape sequence \M
  "Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
/usr/lib/python3/dist-packages/IPython/utils/text.py:597: DeprecationWarning: invalid escape sequence \$
  _dollar_pattern = re.compile("(.*?)\$(\$?[\w\.]+)")
/usr/lib/python3/dist-packages/IPython/core/inputsplitter.py:67: DeprecationWarning: invalid escape sequence \s
  comment_line_re = re.compile('^\s*\#')
/usr/lib/python3/dist-packages/IPython/core/inputtransformer.py:181: DeprecationWarning: invalid escape sequence \)
  """Join lines following explicit line continuations (\)"""
/usr/lib/python3/dist-packages/IPython/core/inputtransformer.py:365: DeprecationWarning: invalid escape sequence \w
  cellmagic_help_re = re.compile('%%\w+\?')
/usr/lib/python3/dist-packages/IPython/core/splitinput.py:50: DeprecationWarning: invalid escape sequence \s
  """, re.VERBOSE)
/usr/lib/python3/dist-packages/IPython/core/splitinput.py:71: DeprecationWarning: invalid escape sequence \s
  pre = re.match('^(\s*)(.*)',line).groups()[0]
/usr/lib/python3/dist-packages/IPython/utils/_tokenize_py3.py:50: DeprecationWarning: invalid escape sequence \s
  cookie_re = re.compile("coding[:=]\s*([-\w.]+)")
/usr/lib/python3/dist-packages/IPython/core/completer.py:946: DeprecationWarning: invalid escape sequence \d
  '''),
/usr/lib/python3/dist-packages/IPython/core/magics/config.py:29: DeprecationWarning: invalid escape sequence \w
  reg = re.compile('^\w+\.\w+$')
/usr/lib/python3/dist-packages/IPython/core/magics/namespace.py:54: DeprecationWarning: invalid escape sequence \?
  re.match('(pinfo )?(\?*)(.*?)(\??$)',parameter_s).groups()
/usr/lib/python3/dist-packages/notebook/notebookapp.py:48: DeprecationWarning: zmq.eventloop.ioloop is deprecated in pyzmq 17. pyzmq now works with default tornado and asyncio eventloops.
  ioloop.install()
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
14:39:22: Debug: ScreenToClient cannot work when toplevel window is not shown
fsleyes_unfiltered  4.24s user 0.80s system 64% cpu 7.838 total
changes on filesystem:                                                          
 fsleyes/gl/__init__.py | 2 +-
 fsleyes/panel.py       | 2 +-

```
</details>

**Comment by pauldmccarthy** [comment]
Cool, so if I add a rule to filter the line after any `DeprecationWarning`, the filtered output will be much cleaner...

**Comment by pauldmccarthy** [comment]
Hopefully this should be fixed in version 0.32.0 - regions below the `lower` threshold are not listed (and stdout filtering should be a bit more robust). Let me know if it works for you.
